dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/deduplication/hibernate/HibernatePotentialDuplicateStore.java

```diff
@@ -125,26 +122,16 @@ private void setStatusParameter( PotentialDuplicateQuery query, Query<?> hiberna
     @SuppressWarnings( "unchecked" )
     public boolean exists( PotentialDuplicate potentialDuplicate )
     {
-        NativeQuery<BigInteger> query;
-        if ( potentialDuplicate.getTeiA() == null )
-        {
-            return false;
-        }
+        if ( potentialDuplicate.getTeiA() == null || potentialDuplicate.getTeiB() == null )
+            throw new PotentialDuplicateException(
+                "Can't search for pair of potential duplicates: teiA and teiB must not be null" );
 
-        if ( potentialDuplicate.getTeiB() == null )
-        {
-            query = getSession().createNativeQuery( "select count(potentialduplicateid) from potentialduplicate pd " +
-                "where pd.teiA = :teia limit 1" );
-            query.setParameter( "teia", potentialDuplicate.getTeiA() );
-        }
-        else
-        {
-            query = getSession().createNativeQuery( "select count(potentialduplicateid) from potentialduplicate pd " +
+        NativeQuery<BigInteger> query = getSession()
+            .createNativeQuery( "select count(potentialduplicateid) from potentialduplicate pd " +
                 "where (pd.teiA = :teia and pd.teiB = :teib) or (pd.teiA = :teib and pd.teiB = :teia) limit 1" );
```

Comment:
una count restituisce già solo una riga, ha senso questo "LIMIT 1" ?

Reply:
yes that is old code thanks for spotting it